cmd/agent/app/agent.go

```diff
@@ -53,18 +57,25 @@ func (a *Agent) Run() error {
 	if err != nil {
 		return err
 	}
+	a.httpAddr.Store(listener.Addr().String())
 	a.closer = listener
 	go func() {
 		if err := a.httpServer.Serve(listener); err != nil {
 			a.logger.Error("http server failure", zap.Error(err))
 		}
+		a.logger.Info("agent's http server exiting")
```

Comment:
wouldn't using a WaitGroup be "cleaner" than doing a log? unless you think this log adds a lot of value outside of code coverage reasons

Reply:
I think log is more useful / functional. WG implies a semantic guarantee of blocking Close(), which is not required here (aside from unit tests).